### ISSUE TITLE
_.omit doesn't support number as keys

### ISSUE THREAD

**Issue by hanax** [jashkenas/underscore#2550]
For example:
```javascript
const obj = {0:"a", 1:"b"};

// returns {0:"a"}
_.pick(obj, 0);
// returns {0:"a"}
_.pick(obj, "0");
// returns {0:"a", 1:"b"}
_.omit(obj, 0);
// returns {1:"b"}
_.omit(obj, "0");
```

I saw this PR https://github.com/jashkenas/underscore/pull/2468 but it doesn't seem very related.
If numbers shouldn't be supported as keys at the first place, at least i think `omit` should have a consistent behaviour as `pick`.

If this does sound like an issue I can submit a PR to try fixing it.

Thank!

**Comment by michaelficarra** [comment]
Looks like a  bug to me. We should `ToPropertyKey` the second parameter. :+1: Looking forward to the PR.

**Comment by hanax** [comment]
@michaelficarra Hmm I found this at work (it is causing a funny bug: showing an error message everywhere on the webpage because of failing to omit it while passing around) and apparently we are not using the latest version. Just found out that has been fixed in the latest version (1.8.3). Thanks!
